src/Microsoft.Azure.WebJobs.Host/Config/ExtensionConfigContext.cs

```diff
@@ -9,6 +12,9 @@ namespace Microsoft.Azure.WebJobs.Host.Config
     /// </summary>
     public class ExtensionConfigContext
```

Comment:
Additions  here are to expose a fluent binding interface that lets us also remove BindingFactory completely.  If we go ahead with this, we just need to add the rest of the Bind* methods here. 
See usage example in test.